File: Sorting/SelectionSort.cpp
#include<bits/stdc++.h>
using namespace std;

void swap(long int&a, long int &b)
{
    long int tmp = a;
    a = b;
    b = tmp;
}

void selection_sort(long int arr[], long int size)
{
    for(int i=0;i<=size-1;i++)
    {
        long int min_index = i;
        for(long int j=i+1;j<=size-1;j++)
        {
            if(arr[j] < arr[min_index])
                min_index = j;
        }
        swap(arr[i], arr[min_index]);
    }
}

int main()
{
long int size, arr[10000];
cin>>size;
for(int i=0;i<=size-1;i++)
    cin>>arr[i];
selection_sort(arr, size);
for(int i=0;i<=size-1;i++)
    cout<<arr[i]<<" ";
}

File: Sorting/SelectionSort.java
import java.util.Arrays;

public class SelectionSort {
	public static void main(String[] args) {
		int[] arr = {1, 2, 5, 3, 4, 6};
		SelectionSort.sort(arr);
		System.out.println(Arrays.toString(arr));
	}

	public static void swap(int[] arr, int index1, int index2) {
		int temp = arr[index1];
		arr[index1] = arr[index2];
		arr[index2] = temp;
	}

	public static void sort(int[] arr) {
		for(int i = 0; i < arr.length; i++) {
			swap(arr, i, findSmallest(arr, i));
		}
	}

	public static int findSmallest(int[] arr, int start) {
		int toReturn = start;

		for(int i = start; i < arr.length; i++) 
			if(arr[i] < arr[toReturn])
				toReturn = i;
		
		return toReturn;
	}
}